apps/dotcom/src/components/IFrameProtector.tsx

```diff
@@ -63,68 +43,20 @@ export function IFrameProtector({
 	context: $ROOM_CONTEXT
 	children: ReactNode
 }) {
-	const [embeddedState, setEmbeddedState] = useState<$EMBEDDED_STATE>(
-		isInIframe() ? EMBEDDED_STATE.IFRAME_UNKNOWN : EMBEDDED_STATE.NOT_IFRAME
-	)
+	const embeddedState = getEmbeddedState(context)
 
 	const url = useUrl()
 
 	useEffect(() => {
-		if (typeof window === 'undefined') {
-			return
-		}
-
-		let timeout: any | undefined
-
-		function handleMessageEvent(event: MessageEvent) {
-			if (!event.source) return
-
-			if (event.data === EXPECTED_QUESTION) {
-				if (!isInIframe()) {
-					// If _we're_ in an iframe, then we don't want to show a nested
-					// iframe, even if we're on a whitelisted host / context
-					event.source.postMessage(EXPECTED_RESPONSE)
-				}
-			}
-
-			if (event.data === EXPECTED_RESPONSE) {
-				// todo: check the origin?
-				setEmbeddedState(EMBEDDED_STATE.IFRAME_OK)
-				clearTimeout(timeout)
-			}
-		}
-
-		window.addEventListener('message', handleMessageEvent, false)
-
-		if (embeddedState === EMBEDDED_STATE.IFRAME_UNKNOWN) {
-			// We iframe embeddings on multiplayer or readonly
-			if (WHITELIST_CONTEXT.includes(context)) {
-				window.parent.postMessage(EXPECTED_QUESTION, '*') // todo: send to a specific origin?
-				timeout = setTimeout(() => {
-					setEmbeddedState(EMBEDDED_STATE.IFRAME_NOT_ALLOWED)
-					trackAnalyticsEvent('connect_to_room_in_iframe', {
-						slug,
-						context,
-						origin: getParentOrigin(),
-					})
-				}, 1000)
-			} else {
-				// We don't allow iframe embeddings on other routes
-				setEmbeddedState(EMBEDDED_STATE.IFRAME_NOT_ALLOWED)
-			}
-		}
-
-		return () => {
-			clearTimeout(timeout)
-			window.removeEventListener('message', handleMessageEvent)
+		if (embeddedState === EMBEDDED_STATE.IFRAME_NOT_ALLOWED) {
+			trackAnalyticsEvent('connect_to_room_in_iframe', {
```

Comment:
I guess this will track more events, as we will also track embedding context that are not whitelisted. But we will have the info to separate the two based on the context and even the parent origin.